lib/board.rb

```diff
@@ -0,0 +1,61 @@
+class Board
+  attr_accessor :cells
+
+  def initialize(cells = nil)
+    if cells == nil
+      @cells = Array.new(9, " ")
+    else
+      @cells = cells
+    end
+  end
+
+  def reset!
+    self.cells = Array.new(9, " ")
+  end
+
+  def display
+    i = 0
+    3.times do 
+      puts " #{self.cells[i]} | #{self.cells[i+1]} | #{self.cells[i+2]} "
+      puts "-----------"
+      i += 3
+    end
+  end
+
+  def position(position)
+    if position.is_a?(String)
+      position = position.to_i - 1 
+    end 
+    
+    self.cells[position]
+  end
+
+  def full?
+    self.cells.all?{|cell| cell == "X" || cell == "O"}
+  end
+
+  def turn_count
+    count = self.cells.count{|cell| cell == "X" || cell == "O"}
```

Comment:
You don't need to store your return value in a local variable; just utilize the implicit return value of the expression.
```
def turn_count
  self.cells.count {|cell| cell == "X" || cell == "O"}
end
```